numpy/fft/_pocketfft.py

```diff
@@ -559,8 +584,9 @@ def hfft(a, n=None, axis=-1, norm=None):
     a = asarray(a)
     if n is None:
         n = (a.shape[axis] - 1) * 2
-    unitary = _unitary(norm)
-    return irfft(conjugate(a), n, axis) * (sqrt(n) if unitary else n)
+    output = irfft(conjugate(a), n, axis)
+    output *= _go_back(n, norm)
+    return output
```

Comment:
you might notice that the hermitian FFTs `hfft` and `ihfft` both use the `_go_back` function, and in a slightly different way; that's just because of the way these transforms are calculated using the corresponding real FFTs, so the scaling factors are also modified accordingly

Reply:
You could also exchange "backward" <-> "forward" and just pass the normalization mode down to `rfft`/`irfft`. Then the normalization is only applied once.

Reply:
yes that'd certainly work but I'd need a 4-case `if` clause: two cases to check if `norm` is `backward` or `forward` so that I can exchange between them as you point out, another case to check if it's `ortho` so that I call the real FFTs with `ortho` (since its "opposite" is itself), and a final `else` to raise a `ValueError` in case `norm` was none of the above. 

Essentially I would do the checks done at `_go_back()`, which would save me one scaling operation as you rightly say. The caveat is that the computation would completely bypass `_go_back()` so all checks must be done within the `hfft`/`ihfft` functions, which would become quite large. Additionally, I thought I'd stay close to the way the computations were done until now. In a nutshell, my thinking was to keep `hfft` & `ihfft` calling `_go_back()` which makes the necessary checks, so that the module is consistent as a whole (easier to maintain & read). 

Do you think that makes sense, or making the checks and saving one array scaling operation sounds like the best way to go here?

Reply:
I guess it's a matter of taste, but I don't think it's really that much code at all:
```python
def _swap_direction(norm):
  try:
    return {"backward": "forward", None: "forward",
            "forward": "backward", "ortho": "ortho"}[norm]
  except KeyError:
    raise ValueError(
      f'Invalid norm value {norm}; should be "backward"'
       '"ortho" or "forward".')
```

Reply:
thanks that's really good! :) 
I didn't think I could write that in such a compact way, I'll make sure to add it